### PR TITLE
Add Block to button_to method

### PR DESCRIPTION
**Pull request by eliasjpr**
This allows to pass a block to a button_to method:

### CODE REVIEW

File: spec/jasper_helpers/links_spec.cr [hunk 1]
```diff
@@ -25,7 +25,56 @@ describe JasperHelpers::Links do
...
+          hidden_field(:_csrf, value: "some value")
+        end
+
+        form.should eq button_to_with_block
+      end
+
+      it "renders form with hidden field" do
+	    form = button_to("Save", "/save", :delete) do
+          	hidden_field(:_csrf, value: "some value") + %(Random text)
+        end
```
Comment (faustinoaq) [inline spec/jasper_helpers/links_spec.cr]: Fix identation

File: spec/jasper_helpers/links_spec.cr [hunk 2]
```diff
@@ -25,7 +25,56 @@ describe JasperHelpers::Links do
     end
 
     it "changes the method when provided" do
-      button_to("Save", "/save", :put).should eq("<form action=\"/save\" class=\"button\" method=\"post\"><input type=\"hidden\" name=\"_method\" id=\"_method\" value=\"put\"/><button type=\"submit\">Save</button></form>")
+      button_to("Save", "/save", :put).should eq button_to_no_block
```
Comment (elorest) [inline spec/jasper_helpers/links_spec.cr]: I like it but feel that the block should be defined in the same scope for readibility unless it's being called by multiple tests.

Comment (elorest) [reply spec/jasper_helpers/links_spec.cr]: expected_result =  <<-FORM
   <form action="save" class="button" method="post">
   <input type="hidden" name="_method" id="_method" value="put"/>
   <button type="submit">Save</button>
   </form>
   FORM

File: src/jasper_helpers/links.cr [hunk 3]
```diff
@@ -26,4 +26,14 @@ module JasperHelpers::Links
       content(:button, body, {:type => "submit"})
     end
   end
+
+  def button_to(body : String, url : String, method = :post)
+    button_content = yield
+    form(action: url, class: "button", method: method) do
+      String.build do |str|
+        str << button_content
+        str << content(:button, body, {:type => "submit"})
```
Comment (elorest) [inline src/jasper_helpers/links.cr]: We beed to apply class to button instead of form so that bootstrap css can style it right. Form doesn't really need to be styled.

File: src/jasper_helpers/links.cr [hunk 4]
```diff
@@ -26,4 +26,14 @@ module JasperHelpers::Links
       content(:button, body, {:type => "submit"})
     end
   end
+
+  def button_to(body : String, url : String, method = :post)
+    button_content = yield
+    form(action: url, class: "button", method: method) do
+      String.build do |str|
+        str << button_content
```
Comment (elorest) [inline src/jasper_helpers/links.cr]: Since you're not doing any checks on the yielded result first it makes more sense to yield directly into the string builder.
`str << yield` This saves time and memory and IMO is more readable.

File: spec/jasper_helpers/links_spec.cr [hunk 5]
```diff
@@ -21,11 +21,52 @@ describe JasperHelpers::Links do
...
+          :delete,
+          "btn btn-primary",
+          "button"
+        )
+      end
     end
   end
 end
+
+def expected_form(content = "", override_method = nil, css_class = nil, type = "submit")
```
Comment (elorest) [inline spec/jasper_helpers/links_spec.cr]: Interesting.

File: src/jasper_helpers/links.cr [hunk 6]
```diff
@@ -15,15 +15,27 @@ module JasperHelpers::Links
     link_to(yield, url, **options)
   end
 
-  def button_to(body : String, url : String, method = :post, **options : Object)
-    form(action: url, class: "button", method: method) do
-      content(:button, body, options)
+  def button_to(body : String, url : String, method : Symbol = :post)
```
Comment (elorest) [inline src/jasper_helpers/links.cr]: Why did you remove options from here?

Comment (eliasjpr) [reply src/jasper_helpers/links.cr]: That got moved down kinda. I reorder the methods to show the progression of the overloading. So it shows as 

Basic usage 
Basic usage with yield 
Basic usage with yield and options
